mapper-processor/src/main/java/com/datastax/oss/driver/internal/mapper/processor/dao/DaoQueryMethodGenerator.java

```diff
@@ -48,25 +51,35 @@ public DaoQueryMethodGenerator(
     nullSavingStrategyValidation = new NullSavingStrategyValidation(context);
   }
 
+  protected Set<DaoReturnTypeKind> getSupportedReturnTypes() {
```

Comment:
Is there a chance that it will be called multiple times?
If so shouldn't we construct it once (possibly in static block) and assign to constant? (only returning it here)

Reply:
We'd still have to expose the protected method in order to allow subclasses to override it, so that wouldn't prevent badly written code to call the method instead of using the constant.

In the end it's all our own internal code so the risk is low. And instantiating the set twice is not catastrophic, just slightly inefficient.